### PR TITLE
Special footer

### PR DESCRIPTION
**Pull request by sarahvc**
Added a special footer option based on this design:
https://uxframework.pearson.com/c/special-application-header#footer

### CODE REVIEW

**Comment by davodey** [comment]
@sarahvc I noticed there is no bg color set to the footer, so when it goes to responsive mode, you see the content underneath.

screen shot 2018-07-24 at 3 58 49 pm

**Comment by davodey** [comment]
@sarahvc Would it be possible to display multiple footers on the demo page?  At least for some of the more prominent options, like the logo and maybe dark mode?  I feel this would help out devs and designers both.

File: src/components/Footer/Footer.js [hunk 1]
```diff
@@ -18,23 +18,44 @@ const Footer = ({ copyrightText, links, anchorTarget, light, singlePageStick })
...
+  const stickCheck = singlePageStick ? '--stick':'';
+  
+    return (
+      <footer className={`pe-footer${stickCheck} pe-label${lightCheck}`}>
+        {renderFooter()}
+      </footer> 
+    );
 }
 
 export default Footer;
```
Comment (davodey) [inline src/components/Footer/Footer.js]: We need to add a propType for the logo.  Probably something like: `logo: PropTypes.bool`

File: src/components/Footer/Footer.js [hunk 2]
```diff
@@ -18,23 +18,44 @@ const Footer = ({ copyrightText, links, anchorTarget, light, singlePageStick })
...
 
-  return (
-      <footer className={`pe-footer${stickCheck} pe-label ${lightCheck}`}>
-        <ul>
-          {renderLinks()}
-        </ul>
-        {renderCopy()}
-      </footer>
-  );
+  const lightCheck = light ? ' pe-footer--light':'';
```
Comment (davodey) [inline src/components/Footer/Footer.js]: This looks good.  If you wanted to trim it down you could by adding a style object.  If your interested, check it out: [https://reactjs.org/docs/dom-elements.html#style](https://reactjs.org/docs/dom-elements.html#style)

File: src/components/Footer/Footer.js (hunk 1 above)
Comment (sarahvc) [reply src/components/Footer/Footer.js]: Added the protoTypes and defaultProp.

**Comment by sarahvc** [comment]
@davodey Thanks Dave! I've pushed another commit, in which I 1) updated the logo version footer after confirmed with UXF designers. 2) added logo into the the protoType, and added default value for logo, 3)Added demo of different options in the demo page, 4)Removed demo footer's singlePageStick props to false.

More about 4): the singlePageStick props is intended for pages that content do not exceed one screen. But the demo page actually exceeds one screen when squeezed to a smaller screen size. So it's not suitable to set this props to true. I was trying to find a solution that we could make the footer work for scenario like this. All methods that I found involves doing something to the content. I think that would be beyond the scope for the footer component. So I left it as it is and updated the props to prompt consumers to check for all screen size before setting this props to true. Let me know if this makes sense.

Thank you!

**Pull request merged**